### PR TITLE
feat: new ctl

### PR DESCRIPTION
**Pull request by hacdias**
Closes #1137

@lidel I'm pretty sure you're more aware of the changes made to this version than I am. Could you give me a hand with this? Also, I noticed `ipfsd.stop` hangs and it does not take timeouts anymore...

### CODE REVIEW

**Comment by lidel** [comment]
Will look into this, but it requires bigger refactor than I thought.
In case  I don't post any updates later today, we will get back to this after holidays.

File: src/daemon/daemon.js [hunk 1]
```diff
@@ -43,18 +41,28 @@ async function cleanup (ipfsd) {
...
-    repoPath: path,
+    test: false,
+    args: flags,
+    type: type
+  })
+
+  const ipfsd = await factory.spawn({
+    ipfsOptions: {
+      repo: path
+    },
```
Comment (lidel) [inline src/daemon/daemon.js]: I believe basic init did not work because repo path got moved to `ipfsOptions`.  This was the fix :)

Comment (hacdias) [reply src/daemon/daemon.js:59]: About the `api` file: I thought IPFS itself handled it! The only thing we did was this: #722.

**Comment by hacdias** [comment]
I'm taking a new look at this on #1411.

**Comment by hacdias** [comment]
I'm closing this in favour of #1411. Thanks for all your work here @lidel. I've reused most of it ;)

**Pull request closed without merging**